### PR TITLE
migrated from RabbitMQ to Kafka (GDEV-1327)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
and update to current event schemas.

Bumps version to 0.3.0.

Co-authored-by: Moritz Hahn <75744178+MoritzHahn1337@users.noreply.github.com>
Co-authored-by: Thomas Zajac <thomas-jakob.zajac@uni-tuebingen.de>
```

### CODE REVIEW

File: example_config.yaml [hunk 1]
```diff
@@ -1,11 +1,18 @@
 aws_config_ini: null
 db_connection_str: '**********'
 db_name: dev_db
+file_registered_event_topic: internal_file_registry
+file_registered_event_type: donwload_served
```
Comment (mephenor) [inline example_config.yaml:5]:
```suggestion
file_registered_event_type: download_served
```

File: ifrs/adapters/inbound/event_sub.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    files_to_register_type: str = Field(
+        ...,
+        description=("The type used for events informing about new files to register."),
+        example="files_to_register",
+    )
+
+
+class EventSubTranslator(EventSubscriberProtocol):
+    """A triple hexagonal translator compatible with the EventSubscriberProtocol that
+    is used to received metadata on new files to register."""
```
Comment (mephenor) [inline ifrs/adapters/inbound/event_sub.py]:
```suggestion
    is used to receive metadata on new files to register."""
```

File: setup.cfg [hunk 3]
```diff
@@ -35,8 +35,8 @@ zip_safe = False
 include_package_data = True
 packages = find:
 install_requires =
-    typer==0.4.1
-    hexkit[s3,mongodb]==0.7.0
+    ghga-event-schemas==0.7.2
+    hexkit[mongodb,s3,akafka]==0.8.0
```
Comment (mephenor) [inline setup.cfg]:
```suggestion
    hexkit[mongodb,s3,akafka]==0.8.1
```

**Review by mephenor — DISMISSED** [review]
Just a quick questiong, why do we have a list of encrypted part sizes now, shouldn't this be uniform?

File: ifrs/container.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    # outbound translators:
+    file_metadata_dao = get_constructor(
+        FileMetadataDaoConstructor, dao_factory=dao_factory
+    )
+    event_publisher = get_constructor(
+        EventPubTranslator, config=config, provider=event_pub_provider
+    )
+
+    # outbound adapters (not following the triple hexagonal translator/provider
```
Comment (MoritzHahn1337) [inline ifrs/container.py]:
```suggestion
    # outbound adapters (not following the triple hexagonal translator/provider)
```

File: ifrs/core/models.py [hunk 5]
```diff
@@ -50,27 +50,27 @@ class FileMetadata(BaseModel):
...
     )
-    encrypted_part_size: int = Field(
+    encrypted_part_sizes: list[int] = Field(
         ...,
         description=(
-            "The size of the file parts of the encrypted content (excluding the"
-            + " crypt4GH) as used for the encryption_parts_md5 and the"
-            + " encryption_parts_sha256 in bytes. The same part size is recommended for"
+            "The sizes of the file parts of the encrypted content (excluding the"
+            + " crypt4GH envelope) as used for the encrypted_parts_md5 and the"
```
Comment (MoritzHahn1337) [inline ifrs/core/models.py]:
```suggestion
            + " crypt4gh envelope) as used for the encrypted_parts_md5 and the"
```

I don't care if we use `crypt4gh` or `crypt4GH`, be would should probably keep it consistent - at least throughout the document.

**Review by MoritzHahn1337 — DISMISSED** [review]
I disregarded everything to do with the `decrypted_part_sizes` as discussed.

Apart from the doc typos, everything else is fine.

**Review by mephenor — DISMISSED** [review]
Nothing to add to Moritz review...

**Review by mephenor — APPROVED** [review]

**Review by MoritzHahn1337 — APPROVED** [review]

**Pull request merged**